waspc/test/AppSpec/FromJSONTest.hs

```diff
@@ -0,0 +1,302 @@
+module AppSpec.FromJSONTest (spec_AppSpecFromJSON) where
+
+import qualified Data.Aeson as Aeson
+import Data.Aeson.Types (FromJSON)
+import Data.Maybe (fromJust)
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as TE
+import NeatInterpolation (trimming)
+import StrongPath (relfileP)
+import Test.Tasty.Hspec
+import Wasp.AppSpec (Ref)
+import qualified Wasp.AppSpec.Action as Action
+import qualified Wasp.AppSpec.App.EmailSender as EmailSender
+import qualified Wasp.AppSpec.Core.Ref as Ref
+import Wasp.AppSpec.Entity (Entity)
+import qualified Wasp.AppSpec.ExtImport as ExtImport
+import qualified Wasp.AppSpec.Job as Job
+import Wasp.AppSpec.Page (Page)
+import qualified Wasp.AppSpec.Page as Page
+import Wasp.AppSpec.Query (Query)
+import qualified Wasp.AppSpec.Query as Query
+
+spec_AppSpecFromJSON :: Spec
+spec_AppSpecFromJSON = do
```

Comment:
I think these test cases cover all JSON tree leaves (`ExtImport`, `Ref`, etc.), as well as some of the inner nodes.

A test for a parent node "trusts" that all children nodes are properly parsed.

This avoids repetition and also gives us nicer failures (if we screw up parsing `ExtImport`, only `ExtImport` tests fail).

Reply:
That makes sense to me!